src/gmail.js

```diff
@@ -1067,7 +1100,57 @@ var Gmail = function(localJQuery) {
         return result;
     };
 
+    api.tools.check_event_type = function(threadObj) {
+        const action_map = {
+            // ""            : "add_to_tasks",
+            "^a"          : "archive",
+            "^k"          : "delete",
+            // ""            : "delete_message_in_thread",
+            // ""            : "delete_forever",
+            // ""            : "delete_label",
+            // ""            : "discard_draft",
+            // ""            : "expand_categories",
+            // ""            : "filter_messages_like_these",
+            // ""            : "label",
+            // ""            : "mark_as_important",
+            // ""            : "mark_as_not_important",
+            // ""            : "mark_as_not_spam",
+            // ""            : "mark_as_spam",
+            // ""            : "move_label",
+            // ""            : "move_to_inbox",
+            // ""            : "mute",
+            // ""            : "read",
+            // ""            : "save_draft",
+            // ""            : "send_message",
+            // ""            : "show_newly_arrived_message",
+            // ""            : "star",
+            // ""            : "undo_send",
+            // ""            : "unmute",
+            // ""            : "unread",
+            // ""            : "unstar",
+            // ""            : "new_email",
+            // ""            : "poll",
+            // ""            : "refresh",
+            // ""            : "restore_message_in_thread",
+            // ""            : "open_email",
+            // ""            : "toggle_threads"
+        };
+        const threadData = api.tools.get_thread_data(threadObj);
+
+        if (threadData && api.check.data.is_action(threadData)) {
+            const action = api.tools.get_action_type(threadData);
+            if (!action_map[action]) {
+                return;
+            } else {
+                return action_map[action];
+            }
+        }
+
+        return;
+    };
```

Comment:
This part has been cleaned up considerably, and I really like that. This is much, much better.

Now though, I see it can be cleaned up even more 😄 

1. I think checking `if (action_map[action])` directly would be better, and avoiding the whole if-else business.
2. That means we only need one additional return, at the end of the function, as a fallback for other cases.
3. I'd also like that final `return`-statement to return `null` for consistency (either all code-paths return something or no-one does)